src/Select/common/SelectWrapper.js

```diff
@@ -0,0 +1,149 @@
+        const { onFocus } = this.props
+
+        if (onFocus) {
+            onFocus(e)
+        }
+    }
+
+    handleToggle = e => {
+        e.stopPropagation()
+
+        this.setState(prevState => ({ open: !prevState.open }))
+    }
+
+    handleClose = () => {
+        this.setState({ open: false })
+    }
+
+    handleOutsideClick = e => {
+        const { onBlur } = this.props
+        const hasRef = this.selectRef.current
+        const isInsideClick =
+            hasRef && this.selectRef.current.contains(e.target)
+
+        if (!isInsideClick) {
+            if (onBlur) {
+                onBlur(e)
+            }
+
+            this.setState({ open: false })
+        }
+    }
+
+    handleKeyPress = e => {
+        e.stopPropagation()
+
+        const { open } = this.state
+        const { keyCode } = e
+        const shouldOpen =
+            !open &&
+            (keyCode === SPACE_KEY ||
+                keyCode === UP_KEY ||
+                keyCode === DOWN_KEY)
+        const shouldClose = open && keyCode === ESCAPE_KEY
+
+        if (shouldClose) {
+            return this.setState({ open: false })
+        }
+
+        if (shouldOpen) {
+            return this.setState({ open: true })
+        }
+    }
+
+    render() {
+        const { open } = this.state
+        const { children, selected, onChange, tabIndex, maxHeight } = this.props
+
+        // Create the input
+        const inputProps = {
+            selected,
+            onChange,
+            options: children,
+        }
+        const input = React.cloneElement(this.props.input, inputProps)
+
+        // Create the menu
+        const menuProps = {
+            selected,
+            onChange,
+            options: children,
+            handleClose: this.handleClose,
+        }
+        const menu = React.cloneElement(this.props.menu, menuProps)
+
+        return (
+            <div
+                className="select"
+                ref={this.selectRef}
+                onFocus={this.handleFocus}
+                onKeyDown={this.handleKeyPress}
+            >
+                <InputWrapper
+                    onToggle={this.handleToggle}
+                    open={open}
+                    tabIndex={tabIndex}
+                >
+                    {input}
+                </InputWrapper>
+                {open && (
+                    <MenuWrapper maxHeight={maxHeight}>{menu}</MenuWrapper>
+                )}
+
+                <style jsx>{`
+                    .select {
+                        position: relative;
+                    }
+                `}</style>
+            </div>
+        )
+    }
+}
+
+SelectWrapper.defaultProps = {
+    tabIndex: '0',
+    maxHeight: '280px',
+}
+
+SelectWrapper.propTypes = {
+    selected: propTypes.any.isRequired,
+    onChange: propTypes.func.isRequired,
+
+    onFocus: propTypes.func,
+    onBlur: propTypes.func,
+
+    children: propTypes.node,
+
+    input: propTypes.element.isRequired,
```

Comment:
To be honest. I would very much prefer if in the organisation of our source and components we stick to simple, lintable rules. It's why I prefer prettier as well. I've spent a lot of time trying to adapt to the style preferences in this repo, and I'm still doing it wrong.

That's time I can't spend on actually building things. I think we should get prettier and eslint to warn us about anything that has to do with style. Having to remind eachother of style preferences in PRs is sth I'd prefer to avoid as much as possible (other than really high-level constructs that can't be caught by linters and tools).

Reply:
Prettier is active though so we shouldn't have any comments based on that config? Agreed with linters. I'd be fine with just sorting them alphabetically automatically.

Reply:
Yeah, I wouldn't mind alphabetical sorting, although I'd still prefer to have required prop types at the top, so it's much easier to see what's required and what not when looking at the code.
It's not really relevant for 2-10 props, but some components have more than that, then it's a bit more difficult to know which ones are required right away. But I don't know if there's a linter for that

Reply:
Sorry if my comment came across a bit snarky btw, don't know if it did. But yeah, I'd agree on the alphabetical sorting. I remember a discussion on sorting css properties alphabetically here (https://github.com/csscomb/csscomb.js/issues/158) where I kind of agree with the points made in favor of alphabetical sorting.

My main point would be to try and codify (and automate) our style preferences as much as possible. If we prefer having `.isRequired` props at the top, I'd want to enforce and fix it via linting. Otherwise it increases the barrier to contributing and means that devs that know the codebase have to police the styles for each PR. To a certain degree that's probably unavoidable, but I'd like to avoid it as much as possible.

I feel the same about our import sorting btw, @varl and I talked about that a while back. I remember we also wanted to use the import eslint plugin for that. Would mean we don't have to worry about that in reviews anymore.

Reply:
Ah @Mohammer5, there's a rule that does exactly what you want: https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/sort-prop-types.md#requiredfirst